### PR TITLE
Serialize table before storing in ethereum

### PR DESCRIPTION
**Pull request by albertolerda**

### CODE REVIEW

File: src/lua/zencode_ethereum.lua [hunk 1]
```diff
@@ -151,7 +151,7 @@ function(obj)
   local content = have(obj)
   local tx = have'ethereum transaction'
   ZEN.assert(not tx.data or #tx.data == 0, "Cannot overwrite transaction data")
-  tx.data = ETH.make_storage_data(content)
+  tx.data = ETH.make_storage_data(ZEN.serialize(content))
```
Comment (jaromil) [inline src/lua/zencode_ethereum.lua:154]: serialize is one way (irreversible) is that enough?

Comment (albertolerda) [reply src/lua/zencode_ethereum.lua:154]: Ah, it is irreversible, no it is not enough...

Comment (jaromil) [reply src/lua/zencode_ethereum.lua:154]: then I guess it should be:
`O.to_base64( O.from_string( JSON.encode( content, 'base64' ) ) )`

Comment (albertolerda) [reply src/lua/zencode_ethereum.lua:154]: Thanks

**Comment by jaromil** [comment]
see PR #323 
```lua
local encoded = MPACK.encode( table )
decoded = MPACK.decode( table )
```
makes sense to also compress the MPACK so I'm working to create a ZPACK too

**Comment by albertolerda** [comment]
Fantastic!

**Comment by jaromil** [comment]
Merging PR now, lets meet over details to rebase this.

**Comment by albertolerda** [comment]
Wait I am doing some changes

**Pull request closed without merging**